plugins/autocomplete/plugin.js

```diff
@@ -727,6 +730,18 @@
 		 */
 		createElement: function() {
 			var el = new CKEDITOR.dom.element( 'ul', this.document );
+			el.setAttributes( {
+				'role': 'listbox',
+				'aria-label': 'autocomp list',
+				'id': 'ac-listbox-list'
+			} );
+			this.editor.editable().setAttributes( {
+				'role': 'combobox',
+				'aria-autocomplete': 'list',
+				'aria-expanded': 'false',
+				'aria-controls': 'ac-listbox-list',
+				'aria-activedescendant': ''
+			} );
```

Comment:
I'm afraid that this approach is a dead end as it changes the role of the editor. Now the whole editable area is presented to user as a combobox, which is very counter-intuitive. Editor's role should remain unchanged.

Reply:
@Comandeer My second thought was to change autocomplete plugin behavior. After the User provide opening characters (like `@` or `[[`) we could add an input to the autocomplete list, and allow to typing continuously filter in this input field. It could look like this:

![](https://user-images.githubusercontent.com/6498567/112304458-ce126c00-8c9d-11eb-9533-56a9e8eb9710.png)

But this is rather unexpected?

Reply:
Yes, it seems like an overkill. And, despite all a11y challenges, it would be probably difficult to implement.

Probably adding only `[aria-controls]` and `[aria-autocomplete]` attributes to editable will be enough and issues with that should be resolved in #4052. The more important issue is the fact that screen readers are not able to read options in the listbox.

Reply:
I've made some sample [https://github.com/ckeditor/ckeditor4/issues/2199#issuecomment-801450208](https://github.com/ckeditor/ckeditor4/issues/2199#issuecomment-801450208) which shows that `not reading listbox options` is caused by different `iframes`. You can manipulate a little with this sample to put listbox inside / outside of the iframe. It is visible even without screen reader - in a11y tools for browsers. Whenever listbox can't be found (because it is in another iframe than combobox) - combobox element don't have any children (in a11y tree).  
  
We can't have listbox is the same iframe as editor (in `wysiwyg` mode) - because this iframe is actually our content (especially if `fullPage: true`). So we should mark this issue as \`wontfix\` (because we don't want to rewrite the iframe and the entire `wysiwyg` mode) / `upstream`(because it fails due to browser inability to find elements outside the same iframe) ?